nemo/collections/asr/data/data_simulation.py

```diff
@@ -1460,3 +1466,901 @@ def _generate_session(self, idx: int, basepath: str, filename: str, enforce_coun
+
+        # get azimuth
+        azimuth = np.arctan2(rel_position[1], rel_position[0]) / np.pi * 180
+        # get elevation
+        elevation = np.arcsin(rel_position[2] / distance) / np.pi * 180
+
+        return distance, azimuth, elevation
+
+    def __str__(self):
+        with np.printoptions(precision=3, suppress=True):
+            desc = f"{type(self)}:\ncenter =\n{self.center}\ncentered positions =\n{self.centered_positions}\nradius = \n{self.radius:.3}\nabsolute positions =\n{self.positions}\ninternal coordinate system =\n{self.internal_cs}\n\n"
+        return desc
+
+    def plot(self, elev=30, azim=-55, mic_size=25):
+        """Plot microphone positions.
+
+        Args:
+            elev: elevation for the view of the plot
+            azim: azimuth for the view of the plot
+            mic_size: size of the microphone marker in the plot
+        """
+        fig = plt.figure()
+        ax = fig.add_subplot(projection='3d')
+
+        # show mic positions
+        for m in range(self.num_mics):
+            # show mic
+            ax.scatter(
+                self.positions[m, 0],
+                self.positions[m, 1],
+                self.positions[m, 2],
+                marker='o',
+                c='black',
+                s=mic_size,
+                depthshade=False,
+            )
+            # add label
+            ax.text(self.positions[m, 0], self.positions[m, 1], self.positions[m, 2], str(m), c='red', zorder=10)
+
+        # show the internal coordinate system
+        ax.quiver(
+            self.center[0],
+            self.center[1],
+            self.center[2],
+            self.internal_cs[:, 0],
+            self.internal_cs[:, 1],
+            self.internal_cs[:, 2],
+            length=self.radius,
+            label='internal cs',
+            normalize=False,
+            linestyle=':',
+            linewidth=1.0,
+        )
+        for dim, label in enumerate(['x′', 'y′', 'z′']):
+            label_pos = self.center + self.radius * self.internal_cs[dim]
+            ax.text(label_pos[0], label_pos[1], label_pos[2], label, tuple(self.internal_cs[dim]), c='blue')
+        try:
+            # Unfortunately, equal aspect ratio has been added very recently to Axes3D
+            ax.set_aspect('equal')
+        except NotImplementedError:
+            logging.warning('Equal aspect ratio not supported by Axes3D')
+        # Set view
+        ax.view_init(elev=elev, azim=azim)
+        # Set reasonable limits for all axes, even for the case of an unequal aspect ratio
+        ax.set_xlim([self.center[0] - self.radius, self.center[0] + self.radius])
+        ax.set_ylim([self.center[1] - self.radius, self.center[1] + self.radius])
+        ax.set_zlim([self.center[2] - self.radius, self.center[2] + self.radius])
+
+        ax.set_xlabel('x/m')
+        ax.set_ylabel('y/m')
+        ax.set_zlabel('z/m')
+        ax.set_title('Microphone positions')
+        ax.legend()
+        plt.show()
```

Comment:
Hmm maybe make this optional, since one may want to use it in notebooks which default to plot at the end of the cell?